tests/baselines/reference/indexerConstraints2.errors.txt

```diff
@@ -37,4 +44,61 @@ tests/cases/compiler/indexerConstraints2.ts(26,5): error TS2413: Numeric index t
         ~~~~~~~~~~~~~~~
 !!! error TS2413: Numeric index type 'A' is not assignable to string index type 'B'.
         [s: string]: B;
+    }
+    
+    
+    type AliasedNumber = number;
+    
+    interface L {
+        [n: AliasedNumber]: A;
+         ~
+!!! error TS1336: An index signature parameter type cannot be a type alias. Use '[index: number]' instead.
+    }
+    
+    type AliasedString = string;
+    
+    interface M {
+        [s: AliasedString]: A;
+         ~
+!!! error TS1336: An index signature parameter type cannot be a type alias. Use '[index: string]' instead.
+    }
+    
+    type AliasedBoolean = boolean;
+    
+    interface N {
+        [b: AliasedBoolean]: A;
+         ~
+!!! error TS1023: An index signature parameter type must be 'string' or 'number'.
+    }
+    
+    type IndexableUnion = "foo" | "bar";
+    
+    interface O {
+        [u: IndexableUnion]: A;
+         ~
+!!! error TS1337: An index signature parameter type cannot be a union type. Use '[K in IndexableUnion]' instead.

```

Comment:
This version of the error message should only appear when the parent is a type alias

Reply:
Given this suggestion https://github.com/Microsoft/TypeScript/pull/20726#discussion_r157842801, is it still applicable?